os/se/ameba/security_ameba_wrapper_tz.c

```diff
@@ -741,6 +741,56 @@ int se_ameba_hal_delete_storage(uint32_t ss_idx)
 	return ret;
 }
 
+#ifdef CONFIG_HW_GCM_ENC
+int se_ameba_hal_gcm_encrypt(hal_data *dec_data, hal_gcm_param *gcm_param, uint32_t key_idx, hal_data *enc_data)
```

Comment:
Hi @jylee9613 
May I know u using which config ? flat_apps or loadable_apps or loadable_ext?

Tmr, I will test with ur given enable config.

Thank you.

Reply:
Hi @jylee9613 

I tested the latest TizenRT commit without AES GCM, and it was able to reproduce the issue with the given config.

`CONFIG_DEBUG_WARN=y
CONFIG_DEBUG_VERBOSE=y
CONFIG_DEBUG_SECURE_ELEMENT_ERROR=y
CONFIG_DEBUG_SECURE_ELEMENT_INFO=y
CONFIG_HW_AES_ENC=y
CONFIG_HW_GCM_ENC=y`

So, it can be concluded that the issue is not related to the AES GCM implementation.
Upon investigation, the issue occurs when both CONFIG_DEBUG_WARN=y and CONFIG_DEBUG_VERBOSE=y are enabled.
If either of these configs is enabled alone, the issue does not reproduce.

Thank You.

Reply:
Hello @ZhenBei-Sin 
Thank you for checking. Then I'll test with enabling one debug option. 

Thank you:)

Reply:
Hi @jylee9613 ,

This is the new findings:
/os/include/debug.h
os/drivers/seclink/seclink_drv.c

Remove the "vdbg" from the "se_register" with using the config that enables all options you provided. 
It is unable to reproduce the issue.

`
int se_register(const char *path, struct sec_lowerhalf_s *lower)
{
	//vdbg("Registering %s\n", path);

	if (!lower) {
		return -1;
	}
`

Therefore, it can be concluded that the issue is related to the debug print statement. 
Please investigate further on this matter.

Thank You.

Reply:
Hi @jylee9613 , as we knew, `lldbg` is used for booting sequence log. How about try to change this to use "lldbg" instead of "vdbg"? Maybe it is able to resolve hang issue?